index.js

```diff
@@ -209,9 +222,26 @@
           }
           return isPropertyDefined('name', config) ? (config.name + '.' + commandName) : commandName;
         };
-
+        
+        var routes = readFromRoute ? $injector.get('$route').routes : { };
         var getUrl = function () {
-          var url = trackUrlParams ? $location.url() : $location.path();
+          // Using ngRoute provided tracking urls
+          var url = $location.url();
+          var trackUrl;
+          Object.keys(routes).forEach(function (key) {
+            var route = routes[key];
+            // Check if url matches this route
+          	if (!("regexp" in route) || !route.regexp.test(url))
```

Comment:
Please update the if blocks to use curly braces to enclose the blocks. That is the convention.

Also, the indentation is a bit wonky and needs to be fixed.